src/components/MapView/Analyser/index.tsx

```diff
@@ -152,43 +122,128 @@ function Analyser({ classes }: AnalyserProps) {
+  };
 
   const runAnalyser = async () => {
     if (!adminBoundariesExtent) {
       return;
     } // hasn't been calculated yet
 
-    if (!selectedDate) {
+    // Detect from issues and throw errors.
+    if (!form.selectedDate) {
       throw new Error('Date must be given to run analysis');
     }
 
-    if (!hazardLayerId || !baselineLayerId) {
+    if (
+      form.hazardLayerId === 'placeholder' ||
+      form.baselineLayerId === 'placeholder'
+    ) {
       throw new Error('Layer should be selected to run analysis');
     }
 
-    const selectedHazardLayer = LayerDefinitions[
-      hazardLayerId
-    ] as WMSLayerProps;
-    const selectedBaselineLayer = LayerDefinitions[
-      baselineLayerId
-    ] as NSOLayerProps;
-
-    const params: AnalysisDispatchParams = {
-      hazardLayer: selectedHazardLayer,
-      baselineLayer: selectedBaselineLayer,
-      date: selectedDate,
-      statistic,
-      extent: adminBoundariesExtent,
-      threshold: {
-        above: parseFloat(aboveThreshold) || undefined,
-        below: parseFloat(belowThreshold) || undefined,
-      },
-    };
-
-    await dispatch(requestAndStoreAnalysis(params));
+    // Build URL string.
+    const analyserShareURL: string = Object.keys(form)
+      .map((prop: string) => {
+        return [prop, form[prop]].map(encodeURIComponent).join('=');
+      })
+      .join('&');
+
+    // Make sure to run teh nalyser onoly when needed.
+    if (activeUrl.current !== analyserShareURL) {
+      // Store the new URL.
+      activeUrl.current = analyserShareURL;
+
+      const selectedHazardLayer = LayerDefinitions[
+        form.hazardLayerId
+      ] as WMSLayerProps;
+      const selectedBaselineLayer = LayerDefinitions[
+        form.baselineLayerId
+      ] as NSOLayerProps;
+
+      const params: AnalysisDispatchParams = {
+        hazardLayer: selectedHazardLayer,
+        baselineLayer: selectedBaselineLayer,
+        date: form.selectedDate,
+        statistic: form.statistic,
+        extent: adminBoundariesExtent,
+        threshold: {
+          above: parseFloat(form.aboveThreshold) || undefined,
+          below: parseFloat(form.belowThreshold) || undefined,
+        },
+      };
+
+      await dispatch(requestAndStoreAnalysis(params));
+
+      // Set share URL into the browser.
+      history.replace(`?share=true&${analyserShareURL}`);
+    }
   };
 
+  useEffect(() => {
+    // Early return if data not loaded.
+    if (!boundaryLayerData || !availableDates) {
+      return;
+    }
+
+    const {
+      hazardLayerParamId,
+      baselineLayerParamId,
+      selectedParamDate,
+      statisticParam,
+      aboveThresholdParam,
+      belowThresholdParam,
+      fromURL,
+    } = extractPropsFromURL(history.location.search);
+
+    // Set data from the URL.
+    if (fromURL) {
+      // Correct the statistic if invalid.
+      let statisticCorrected: AggregationOperations =
+        AggregationOperations.Mean;
+
+      if (isStatistic(statisticParam)) {
```

Comment:
you shouldn't need to cast if `isStatistic` is made to be a type predicate

Reply:
Done